tests/test_consdbClient.py

```diff
@@ -0,0 +1,140 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import pytest
+import responses
+
+from lsst.summit.utils import ConsDbClient
+
+
+@pytest.fixture
+def client():
+    return ConsDbClient("http://example.com/consdb")
+
+
+def test_table_name(client):
+    instrument = "latiss"
+    obs_type = "exposure"
+    assert client.compute_flexible_metadata_table_name(instrument, obs_type) == "cdb_latiss.exposure_flexdata"
+
+
+@responses.activate
+def test_add_flexible_metadata_key(client):
+    instrument = "latiss"
+    obs_type = "exposure"
+    responses.post(
+        "http://example.com/consdb/flex/latiss/exposure/addkey",
+        json={
+            "message": "Key added to flexible metadata",
+            "key": "foo",
+            "instrument": "latiss",
+            "obs_type": "exposure",
+        },
+        match=[
+            responses.matchers.json_params_matcher({"key": "foo", "dtype": "bool", "doc": "bool key"}),
+        ],
+    )
+    responses.post(
+        "http://example.com/consdb/flex/latiss/exposure/addkey",
+        json={
+            "message": "Key added to flexible metadata",
+            "key": "bar",
+            "instrument": "latiss",
+            "obs_type": "exposure",
+        },
+        match=[
+            responses.matchers.json_params_matcher({"key": "bar", "dtype": "int", "doc": "int key"}),
+        ],
+    )
+    responses.post(
+        "http://example.com/consdb/flex/bad_instrument/exposure/addkey",
+        status=404,
+        json={"message": "Unknown instrument", "value": "bad_instrument", "valid": ["latiss"]},
+    )
+    responses.post(
+        "http://example.com/consdb/flex/latiss/bas_obs_type/addkey",
+        status=404,
+        json={"message": "Unknown observation type", "value": "bad_obs_type", "valid": ["exposure"]},
+    )
+
+    assert client.add_flexible_metadata_key(instrument, obs_type, "foo", "bool", "bool key")["key"] == "foo"
+    assert client.add_flexible_metadata_key(instrument, obs_type, "bar", "int", "int key")["key"] == "int"
+    client.add_flexible_metadata_key("bad_instrument", obs_type, "error", "int", "instrument error")
+    client.add_flexible_metadata_key(instrument, "bad_obs_type", "error", "int", "obs_type error")
+
+
+@responses.activate
+def test_get_flexible_metadata_keys(client):
+    description = [["foo", "bool", "a"], ["bar", "int", "b"]]
+    responses.get(
+        "http://example.com/consdb/flex/latiss/exposure/schema",
+        json=description,
+    )
+    instrument = "latiss"
+    obs_type = "exposure"
+    assert client.get_flexible_metadata_keys(instrument, obs_type) == description
+
+
+@responses.activate
+def test_get_flexible_metadata(client):
```

Comment:
Great to see these types being checked, but I fear that because of how the mocking is done, this doesn't actually test the behaviour of the DB. I guess that's probably all that can be done without significant extra work, but just noting that if it's possible to get something a little more thorough up once we have the real thing that would be good (though I definitely understand there will be reluctance to using `vcr` for that, but perhaps there's something else we can do there).

Reply:
Indeed, these tests don't actually test the server behavior, just the request generation and response formatting.  The list of `responses` is I think effectively a `vcr` cassette, isn't it?